src/components/lib/kernel/cos_kernel_api.c

```diff
@@ -1062,7 +1062,7 @@ cos_rcv(arcvcap_t rcv, rcv_flags_t flags, int *rcvd)
 }
 
 vaddr_t
-cos_mem_aliasn(struct cos_compinfo *dstci, struct cos_compinfo *srcci, vaddr_t src, size_t sz)
+cos_mem_aliasn(struct cos_compinfo *dstci, struct cos_compinfo *srcci, vaddr_t src, size_t sz, unsigned long newflags)
```

Comment:
@gparmer do we need a new name for this `newflags`? I feel that this name is not that easy to understand its meaning

Reply:
Good comment.  `perm_flags`? I think that most would associated `perm` with "permissions"?

Reply:
Agreed, made the change.